src/components/TodoList/TodoList.tsx

```diff
@@ -1,210 +1,115 @@
-            </td>
-
-            <td className="has-text-right is-vcentered">
-              <button data-cy="selectButton" className="button" type="button">
-                <span className="icon">
-                  <i className="far fa-eye-slash" />
-                </span>
-              </button>
-            </td>
-          </tr>
-
-          <tr data-cy="todo">
-            <td className="is-vcentered">4</td>
-            <td className="is-vcentered"><span className="icon" data-cy="iconCompleted"><i className="fas fa-check" /></span></td>
-            <td className="is-vcentered is-expanded">
-              <p className="has-text-success">et porro tempora</p>
-            </td>
-
-            <td className="has-text-right is-vcentered">
-              <button data-cy="selectButton" className="button" type="button">
-                <span className="icon">
-                  <i className="far fa-eye" />
-                </span>
-              </button>
-            </td>
-          </tr>
-
-          <tr data-cy="todo">
-            <td className="is-vcentered">5</td>
-            <td className="is-vcentered"> </td>
+  const dispatch = useAppDispatch();
 
-            <td className="is-vcentered is-expanded">
-              <p className="has-text-danger">laboriosam mollitia et enim quasi adipisci quia provident illum</p>
-            </td>
+  const selectedTodoId = useAppSelector(state => state.currentTodo?.id);
+  const isSelectedTodo = (id: number) => selectedTodoId === id;
+  const handleSelect = (todo: Todo) => dispatch(currentTodoActions.setTodo(todo));
 
-            <td className="has-text-right is-vcentered">
-              <button data-cy="selectButton" className="button" type="button">
-                <span className="icon">
-                  <i className="far fa-eye" />
-                </span>
-              </button>
-            </td>
-          </tr>
+  const visableTodos = useAppSelector(state => {
+    const { todos, filter: filterState } = state;
 
-          <tr data-cy="todo">
-            <td className="is-vcentered">6</td>
-            <td className="is-vcentered"> </td>
+    return todos.filter(todo => {
+      switch (filterState.status) {
+        case 'active':
+          return !todo.completed;
 
-            <td className="is-vcentered is-expanded">
-              <p className="has-text-danger">qui ullam ratione quibusdam voluptatem quia omnis</p>
-            </td>
+        case 'completed':
+          return todo.completed;
 
-            <td className="has-text-right is-vcentered">
-              <button data-cy="selectButton" className="button" type="button">
-                <span className="icon">
-                  <i className="far fa-eye" />
-                </span>
-              </button>
-            </td>
-          </tr>
+        default:
+          return todo;
+      }
+    }).filter(todo => todo.title.toLowerCase().includes(filterState.query.toLowerCase()));
```

Comment:
you will call `filterState.query.toLowerCase()` on every iteration
better to do it ones and pass variable in callback then